schemas/input/csl-citation.json

```
@@ -35,122 +35,262 @@
-        "additionalProperties": false
-      }
-    },
-    "properties": {
-      "type": "object",
-      "properties": {
-        "noteIndex": {
-          "type": "number"
+          "minItems": 1,
+          "additionalProperties": false
+        }
+      },
+      "suppress-author": {
+        "type": ["string", "number", "boolean"]
+      },
+      "author-only": {
+        "type": ["string", "number", "boolean"]
+      },
+      "uris": {
+        "type": "array",
+        "items": {
+          "type": "string"
         }
       },
+      "required": ["id"],
       "additionalProperties": false
     }
   },
-  "required": ["schema", "citationID"],
-  "additionalProperties": false,
-  "definitions": {
-    "locators": {
-      "properties": {
-        "appendix": {
-          "type": "string"
-        },
-        "article": {
-          "type": "string"
-        },
-        "book": {
-          "type": "string"
-        },
-        "canon": {
-          "type": "string"
-        },
-        "chapter": {
-          "type": "string"
-        },
-        "column": {
-          "type": "string"
-        },
-        "elocation": {
-          "type": "string"
-        },
-        "equation": {
-          "type": "string"
-        },
-        "figure": {
-          "type": "string"
-        },
-        "folio": {
-          "type": "string"
-        },
-        "issue": {
-          "type": "string"
-        },
-        "line": {
-          "type": "string"
-        },
-        "note": {
-          "type": "string"
-        },
-        "opus": {
-          "type": "string"
-        },
-        "page": {
-          "type": "string"
-        },
-        "paragraph": {
-          "type": "string"
-        },
-        "part": {
-          "type": "string"
-        },
-        "rule": {
-          "type": "string"
-        },
-        "section": {
-          "type": "string"
-        },
-        "sub-verbo": {
-          "type": "string"
-        },
-        "supplement": {
-          "type": "string"
-        },
-        "table": {
-          "type": "string"
-        },
-        "timestamp": {
-          "type": "string"
-        },
-        "title": {
-          "type": "string"
-        },
-        "verse": {
-          "type": "string"
+  "properties": {
+    "type": "object",
+    "properties": {
+      "noteIndex": {
+        "type": "number"
+      }
+    },
+    "additionalProperties": false
+  }
+},
+"required": ["schema", "citationID"],
+"additionalProperties": false,
+"definitions": {
+  "locator-variable": {
+    "description": "The value of a locator can be an array of numbers and/or strings OR a range object.",
+    "type": "array",
+    "items": {
+      "anyOf": [{
+          "type": ["number", "string"]
         },
-        "volume": {
-          "type": "string"
+        {
+          "title": "Locator Range",
+          "type": "object",
+          "properties": {
+            "begin": {
+              "type": ["number", "string"]
```

```suggestion
              "type": "#/definitions/locator-variable"
```

This needs to be a `locator-variable` object. Right now, a range object doesn't specify the type of locator it is. Changing the definition also supports compound ranges.